File: java-advent-of-code/src/test/java/advent/twenty_seventeen/Day15DuelingGeneratorsTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day15DuelingGeneratorsTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day15DuelingGenerators("/twenty_seventeen/Day15-DuelingGenerators-input.txt");

        assertEquals(612, dp.getPart1Answer().intValue());
        assertEquals(285, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day15DuelingGenerators("/twenty_seventeen/Day15-DuelingGenerators-example#1.txt");

        assertEquals(588, dp.getPart1Answer().intValue());
        assertEquals(309, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day15DuelingGeneratorsTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day15DuelingGeneratorsTest  extends FlatSpec with Matchers {
  "2017 Day 15 - Input" should "calculate answers" in {
    val dp = new Day15DuelingGenerators("/twenty_seventeen/Day15-DuelingGenerators-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day15DuelingGenerators("/twenty_seventeen/Day15-DuelingGenerators-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
